lib/reek/report/code_climate/code_climate_fingerprint.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module Reek
+  module Report
+    # Generates a string to uniquely identify a smell
+    class CodeClimateFingerprint
+      NON_IDENTIFYING_PARAMETERS = [:count, :depth].freeze
+
+      def initialize(warning)
+        @warning = warning
+      end
+
+      def compute
+        return unless warning_uniquely_identifiable?
+
+        identify_warning
+
+        identifying_aspects.hexdigest
```

Comment:
Would there be any value in freezing the returned digest string for good measure? 😀